LibreNMS/Modules/Wireless.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Wireless.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @link       http://librenms.org
+ *
+ * @copyright  2021 Otto Reinikainen
+ * @author     Otto Reinikainen <otto@ottorei.fi>
+ */
+
+namespace LibreNMS\Modules;
+
+use App\Observers\ModuleModelObserver;
+use Illuminate\Support\Collection;
+use LibreNMS\DB\SyncsModels;
+use LibreNMS\Device\WirelessSensor;
+use LibreNMS\Interfaces\Module;
+use LibreNMS\Interfaces\Polling\WirelessAccessPointPolling;
+use LibreNMS\OS;
+use LibreNMS\RRD\RrdDefinition;
+
+class Wireless implements Module
+{
+    use SyncsModels;
+
+    /**
+     * Discover this module. Heavier processes can be run here
+     * Run infrequently (default 4 times a day)
+     *
+     * @param  OS  $os
+     */
+    public function discover(OS $os)
+    {
+        // Discover wireless sensors
+        WirelessSensor::runDiscovery($os);
+    }
+
+    /**
+     * Poll data for this module and update the DB / RRD.
+     * Try to keep this efficient and only run if discovery has indicated there is a reason to run.
+     * Run frequently (default every 5 minutes)
+     *
+     * @param  OS  $os
+     */
+    public function poll(OS $os)
+    {
+        echo "\nPoll Wireless sensors: ";
+        WirelessSensor::poll($os);
+
+        if ($os instanceof WirelessAccessPointPolling) {
+            echo "\nPoll Wireless Access Points: ";
+            $access_points = new Collection;
+
+            // Get APs from controller. Uniquely identify by CompositeKey (apmac+radionumber)
+            // Each logical unit has it's own record
+            $access_points = $os->pollWirelessAccessPoints()->keyBy(function ($item) {
+                return $item->getCompositeKey();
+            });
+
+            // Sync models. In a situation where controller is changed, update existing APs
+            // TODO: Test restoring soft deleted models
+            ModuleModelObserver::observe('\App\Models\AccessPoint');
+            $this->syncModels($os->getDevice(), 'accessPoints', $access_points);
+
+            // Update RRD-files for AccessPoints and calculate total counters for the controller
+            foreach ($access_points as $ap) {
+                $rrd_name = [$os->getAccessPointDatastorePrefix(),  $ap->name . $ap->radionum];
```

Comment:
I suspect we should try to make common RRD files and point the uniquely named graph files at those.

Reply:
> 
> 
> I suspect we should try to make common RRD files and point the uniquely named graph files at those.

What kind of approach you had in mind? I tried to use the same name for the files in order to preserve existing graphs.

Reply:
Look for  `// enable migration` in #13337 rrd can use another file as a data source during creation.  Maybe we should implement that in the existing code.

Reply:
Sure. The existing files should be migrated and named correctly. I noticed that the Cisco WLC uses the prefix "arubaap" in the code but does not actually save any graphs.

Reply:
> 

Just notices something interesting - Aruba, Cisco and Huawei are currently using the same prefixes for the RRD-files. Should I migrate the existing data or just continue to use the same naming convention?